sysret_test can't be built on arm64

```
# bazel test //test/syscalls:sysret_test_runsc_ptrace
Starting local Bazel server and connecting to it...
WARNING: /root/.cache/bazel/_bazel_root/a5240a713337f9902e76feccf3589eba/external/remote_java_tools_linux/BUILD:671:1: in hdrs attribute of cc_library rule @remote_java_tools_linux//:combiners: Artifact 'external/remote_java_tools_linux/java_tools/src/tools/singlejar/zip_headers.h' is duplicated (through '@remote_java_tools_linux//:transient_bytes' and '@remote_java_tools_linux//:zip_headers'). Since this rule was created by the macro 'cc_library', the error might have been caused by the macro implementation
INFO: Analyzed target //test/syscalls:sysret_test_runsc_ptrace (265 packages loaded, 10705 targets configured).
INFO: Found 1 test target...
INFO: Deleting stale sandbox base /root/.cache/bazel/_bazel_root/a5240a713337f9902e76feccf3589eba/sandbox
ERROR: /home/xuhaibo/go/src/gvisor.dev/gvisor/test/syscalls/linux/BUILD:3328:1: C++ compilation of rule '//test/syscalls/linux:sysret_test' failed (Exit 1) gcc failed: error executing command /usr/local/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer '-std=c++0x' -MD -MF ... (remaining 52 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from external/com_google_googletest/googletest/include/gtest/gtest.h:375:0,
                 from test/syscalls/linux/sysret.cc:20:
test/syscalls/linux/sysret.cc: In member function 'virtual void gvisor::testing::{anonymous}::SysretTest::SetUp()':
test/syscalls/linux/sysret.cc:54:24: error: 'PTRACE_GETREGS' was not declared in this scope
     ASSERT_THAT(ptrace(PTRACE_GETREGS, pid, 0, &regs_), SyscallSucceeds());
                        ^
test/syscalls/linux/sysret.cc:54:24: note: suggested alternative: 'PTRACE_GETREGSET'
test/syscalls/linux/sysret.cc: In member function 'void gvisor::testing::{anonymous}::SysretTest::SetRip(uint64_t)':
test/syscalls/linux/sysret.cc:64:11: error: 'struct user_regs_struct' has no member named 'rip'
     regs_.rip = newrip;
           ^~~
In file included from external/com_google_googletest/googletest/include/gtest/gtest.h:375:0,
                 from test/syscalls/linux/sysret.cc:20:
test/syscalls/linux/sysret.cc:65:24: error: 'PTRACE_SETREGS' was not declared in this scope
     ASSERT_THAT(ptrace(PTRACE_SETREGS, child_, 0, &regs_), SyscallSucceeds());
                        ^
test/syscalls/linux/sysret.cc:65:24: note: suggested alternative: 'PTRACE_SETREGSET'
test/syscalls/linux/sysret.cc: In member function 'void gvisor::testing::{anonymous}::SysretTest::SetRsp(uint64_t)':
test/syscalls/linux/sysret.cc:69:11: error: 'struct user_regs_struct' has no member named 'rsp'; did you mean 'sp'?
     regs_.rsp = newrsp;
           ^~~
           sp
In file included from external/com_google_googletest/googletest/include/gtest/gtest.h:375:0,
                 from test/syscalls/linux/sysret.cc:20:
test/syscalls/linux/sysret.cc:70:24: error: 'PTRACE_SETREGS' was not declared in this scope
     ASSERT_THAT(ptrace(PTRACE_SETREGS, child_, 0, &regs_), SyscallSucceeds());
                        ^
test/syscalls/linux/sysret.cc:70:24: note: suggested alternative: 'PTRACE_SETREGSET'
Target //test/syscalls:sysret_test_runsc_ptrace failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 34.360s, Critical Path: 3.89s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
//test/syscalls:sysret_test_runsc_ptrace                        FAILED TO BUILD

FAILED: Build did NOT complete successfully
```

--- a/test/syscalls/linux/BUILD
+++ b/test/syscalls/linux/BUILD
@@ -3336,10 +3336,7 @@ cc_binary(
 cc_binary(
     name = "sysret_test",
     testonly = 1,
-    srcs = select_arch(
-        amd64 = ["sysret.cc"],
-        arm64 = [],
-    ),
+    srcs = ["sysret.cc"],
     linkstatic = 1,
     deps = [
         gtest,

--- a/test/syscalls/linux/sysret.cc
+++ b/test/syscalls/linux/sysret.cc
@@ -14,6 +14,8 @@
 
 // Tests to verify that the behavior of linux and gvisor matches when
 // 'sysret' returns to bad (aka non-canonical) %rip or %rsp.
+
+#include <linux/elf.h>
 #include <sys/ptrace.h>
 #include <sys/user.h>
 
@@ -32,6 +34,7 @@ constexpr uint64_t kNonCanonicalRsp = 0xFFFF000000000000;
 class SysretTest : public ::testing::Test {
  protected:
   struct user_regs_struct regs_;
+  struct iovec iov;
   pid_t child_;
 
   void SetUp() override {
@@ -48,10 +51,14 @@ class SysretTest : public ::testing::Test {
 
     // Parent.
     int status;
+    memset(&iov, 0, sizeof(iov));
     ASSERT_THAT(pid, SyscallSucceeds());  // Might still be < 0.
     ASSERT_THAT(waitpid(pid, &status, 0), SyscallSucceedsWithValue(pid));
     EXPECT_TRUE(WIFSTOPPED(status) && WSTOPSIG(status) == SIGSTOP);
-    ASSERT_THAT(ptrace(PTRACE_GETREGS, pid, 0, &regs_), SyscallSucceeds());
+
+    iov.iov_base = &regs_;
+    iov.iov_len = sizeof(regs_);
+    ASSERT_THAT(ptrace(PTRACE_GETREGSET, pid, NT_PRSTATUS, &iov), SyscallSucceeds());
 
     child_ = pid;
   }
@@ -61,13 +68,25 @@ class SysretTest : public ::testing::Test {
   }
 
   void SetRip(uint64_t newrip) {
+#if defined(__x86_64__)
     regs_.rip = newrip;
-    ASSERT_THAT(ptrace(PTRACE_SETREGS, child_, 0, &regs_), SyscallSucceeds());
+#elif defined(__aarch64__)
+    regs_.pc = newrip;
+#else
+#error "Unknown architecture"
+#endif
+    ASSERT_THAT(ptrace(PTRACE_SETREGSET, child_, NT_PRSTATUS, &iov), SyscallSucceeds());
   }
 
   void SetRsp(uint64_t newrsp) {
+#if defined(__x86_64__)
     regs_.rsp = newrsp;
-    ASSERT_THAT(ptrace(PTRACE_SETREGS, child_, 0, &regs_), SyscallSucceeds());
+#elif defined(__aarch64__)
+    regs_.sp = newrsp;
+#else
+#error "Unknown architecture"
+#endif
+    ASSERT_THAT(ptrace(PTRACE_SETREGSET, child_, NT_PRSTATUS, &iov), SyscallSucceeds());
   }
 
   // Wait waits for the child pid and returns the exit status.
@@ -104,8 +123,15 @@ TEST_F(SysretTest, BadRsp) {
   SetRsp(kNonCanonicalRsp);
   Detach();
   int status = Wait();
+#if defined(__x86_64__)
   EXPECT_TRUE(WIFSIGNALED(status) && WTERMSIG(status) == SIGBUS)
       << "status = " << status;
+#elif defined(__aarch64__)
+  EXPECT_TRUE(WIFSIGNALED(status) && WTERMSIG(status) == SIGSEGV)
+      << "status = " << status;
+#else
+#error "Unknown architecture"
+#endif
 }
 }  // namespace
 

